drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/ScenarioCommandRegistry.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.commands;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.NoSuchElementException;
+
+import javax.enterprise.context.Dependent;
+
+import org.drools.workbench.screens.scenariosimulation.client.commands.actualcommands.AbstractScenarioSimulationCommand;
+import org.kie.workbench.common.command.client.registry.command.CommandRegistryImpl;
+
+/**
+ * This class is used to store <code>Queue</code>es of <b>executed/undone</b> <code>Command</code>s
+ */
+@Dependent
+public class ScenarioCommandRegistry extends CommandRegistryImpl<AbstractScenarioSimulationCommand> {
+
+    protected final Deque<AbstractScenarioSimulationCommand> undoneCommands = new ArrayDeque<>();
```

Comment:
I think this should be moved to `CommandRegistryImpl` (or at least create a `RedoableCommandRegistry` interface or similar in the common project)

In any case I think it is not working because as soon as a new command is executed, `undoneCommands` should be cleaned up:
**Correct**
1. do `Command1`
2. undo() -> `Command1`
3. redo() -> `Command1`

**Wrong**
1. do `Command1`
2. undo() -> `Command1`
3. do() `Command2`
4. redo() -> `Command1` (wrong)

Reply:
@danielezonca 
Done/Undo will be implemented in DROOLS-3347